core/amber/src/main/scala/edu/uci/ics/texera/web/TexeraCompilerWebApplication.scala

```diff
@@ -0,0 +1,85 @@
+package edu.uci.ics.texera.web
+
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.github.toastshaman.dropwizard.auth.jwt.JwtAuthFilter
+import com.typesafe.scalalogging.LazyLogging
+import edu.uci.ics.amber.engine.common.AmberConfig
+import edu.uci.ics.texera.Utils
+import edu.uci.ics.texera.web.TexeraWebApplication.parseArgs
+import edu.uci.ics.texera.web.auth.JwtAuth.jwtConsumer
+import edu.uci.ics.texera.web.auth.{
+  GuestAuthFilter,
+  SessionUser,
+  UserAuthenticator,
+  UserRoleAuthorizer
+}
+import edu.uci.ics.texera.web.resource.WorkflowCompilationResource
+import io.dropwizard.auth.{AuthDynamicFeature, AuthValueFactoryProvider}
+import io.dropwizard.setup.{Bootstrap, Environment}
+import org.eclipse.jetty.server.session.SessionHandler
+import org.glassfish.jersey.server.filter.RolesAllowedDynamicFeature
+
+object TexeraCompilerWebApplication {
+  def main(args: Array[String]): Unit = {
+    val argMap = parseArgs(args)
+
+    new TexeraCompilerWebApplication().run(
+      "server",
+      Utils.amberHomePath
+        .resolve("src")
+        .resolve("main")
+        .resolve("resources")
+        .resolve("texera-compiler-web-config.yml")
+        .toString
+    )
+  }
+}
+
+class TexeraCompilerWebApplication
+    extends io.dropwizard.Application[TexeraCompilerWebConfiguration]
+    with LazyLogging {
+  override def initialize(bootstrap: Bootstrap[TexeraCompilerWebConfiguration]): Unit = {
+    // register scala module to dropwizard default object mapper
+    bootstrap.getObjectMapper.registerModule(DefaultScalaModule)
+  }
+
+  override def run(
+      configuration: TexeraCompilerWebConfiguration,
+      environment: Environment
+  ): Unit = {
+    // serve backend at /api/texera
+    environment.jersey.setUrlPattern("/api/texera/*")
+
+    // register SessionHandler
+    environment.jersey.register(classOf[SessionHandler])
+    environment.servlets.setSessionHandler(new SessionHandler)
+    environment.jersey.register(classOf[WorkflowCompilationResource])
+
+    if (AmberConfig.isUserSystemEnabled) {
+      // register JWT Auth layer
+      environment.jersey.register(
+        new AuthDynamicFeature(
+          new JwtAuthFilter.Builder[SessionUser]()
+            .setJwtConsumer(jwtConsumer)
+            .setRealm("realm")
+            .setPrefix("Bearer")
+            .setAuthenticator(UserAuthenticator)
```

Comment:
I'd suggest remove authenticator. this service should just authorize the access, it should not authenticate the login.

Reply:
removed

Reply:
If removed it, the access role will not be able to be retrieved, as I believe the `UserAuthenticator` is retrieving information from the jwt token and passing them to the `UserRoleAuthorizer`. And user's request with valid jwt token will be considered as unauthorized. 

So I added it back.

Reply:
indeed. this module should have been named as `JWTAuthenticator` to better justify its purpose. I agree we can keep it for now.